apps/web/app/(app)/environments/[environmentId]/surveys/lib/surveys.ts

```diff
@@ -0,0 +1,216 @@
+import "server-only";
+import { TSurveyListItem } from "@/app/(app)/environments/[environmentId]/surveys/types/survey";
+import { Prisma } from "@prisma/client";
+import { cache as reactCache } from "react";
+import { prisma } from "@formbricks/database";
+import { cache } from "@formbricks/lib/cache";
+import { surveyCache } from "@formbricks/lib/survey/cache";
+import { getInProgressSurveyCount } from "@formbricks/lib/survey/service";
+import { buildOrderByClause, buildWhereClause } from "@formbricks/lib/survey/utils";
+import { validateInputs } from "@formbricks/lib/utils/validate";
+import { ZOptionalNumber } from "@formbricks/types/common";
+import { ZId } from "@formbricks/types/common";
+import { DatabaseError } from "@formbricks/types/errors";
+import { TSurveyFilterCriteria } from "@formbricks/types/surveys/types";
+
+export const selectSurveyListItem: Prisma.SurveySelect = {
+  id: true,
+  createdAt: true,
+  updatedAt: true,
+  name: true,
+  type: true,
+  creator: {
+    select: {
+      name: true,
+    },
+  },
+  status: true,
+  singleUse: true,
+  environmentId: true,
+  _count: {
+    select: { responses: true },
+  },
+};
+
+export const getSurveyListItems = reactCache(
+  (
+    environmentId: string,
+    limit?: number,
+    offset?: number,
+    filterCriteria?: TSurveyFilterCriteria
+  ): Promise<TSurveyListItem[]> =>
+    cache(
+      async () => {
+        validateInputs([environmentId, ZId], [limit, ZOptionalNumber], [offset, ZOptionalNumber]);
+
+        try {
+          if (filterCriteria?.sortBy === "relevance") {
+            // Call the sortByRelevance function
+            return await getSurveyListItemsSortedByRelevance(
+              environmentId,
+              limit,
+              offset ?? 0,
+              filterCriteria
+            );
+          }
+
+          // Fetch surveys normally with pagination and include response count
+          const surveysPrisma = await prisma.survey.findMany({
+            where: {
+              environmentId,
+              ...buildWhereClause(filterCriteria),
+            },
+            select: selectSurveyListItem,
+            orderBy: buildOrderByClause(filterCriteria?.sortBy),
+            take: limit,
+            skip: offset,
+          });
+
+          return surveysPrisma.map((survey) => {
+            return {
+              ...survey,
+              responseCount: survey._count.responses,
+            };
+          });
+        } catch (error) {
+          if (error instanceof Prisma.PrismaClientKnownRequestError) {
+            console.error(error);
+            throw new DatabaseError(error.message);
+          }
+          throw error;
+        }
+      },
+      [`getSurveyListItems-${environmentId}-${limit}-${offset}-${JSON.stringify(filterCriteria)}`],
+      {
+        tags: [surveyCache.tag.byEnvironmentId(environmentId)],
```

Comment:
since we also fetch the responses we also need to make this depend on the responseCache. Otherwise the response count would stay the same even though new responses come in.